### DISCUSSION TITLE
Creating Listings with Materials Etsy API V3

### DISCUSSION THREAD

**Discussion by JHarding86** [etsy/open-api#1113] [Q&A] [answered]
I have been using the Etsy API for a couple of weeks now, everything seems to be working well... except for one thing.

While using the createDraftListing call I am adding an array of materials just as specified by the API.

 

For example, when I use this array in my call, only the last string in my array is added to the listing ("creativity").

["wood", "laser", "ribbon", "personalization", "hard work", "dedication", "craftsmanship", "Hand Finished", "laser engraved", "Christmas", "Ornament", "birch", "creativity"]
 
Any idea why only the last item is being added?

**Comment by emergingdzns** [comment]
I *think* that when you submit the materials, it says "array" but really it's likely a comma delimited single string like:
"wood,laser,ribbon,personalization,hard work,dedication,craftsmanship,Hand Finished,laser engraved,Christmas,Ornament,birch,creativity"

**Reply by JHarding86** [reply]
I will give this a try.

I _know_ when requesting the materials from an existing listing, the Etsy API returns an array. That's neither here nor there.

**Reply by seandmccarthy** [reply]
I can confirm that when you call createDraftListing each of tags, materials and styles all need to be supplied as a CSV string, despite being listed as "Array" type.

**Reply by JHarding86** [reply, marked as answer]
I also confirmed this a few days ago.
